### PR TITLE
VideoPlayer: collect runtime and platform information

### PR DESCRIPTION
**Pull request by FernetMenta**
@popcornmix this is supposed to solve the issue related to non-existent fallback setting for deinterlacing.  Platforms like Pi want halfed, others full.

It also introduces a new concept for doing platform ports without ugly ifdefs and the need for touching mainstream code. The class CProcessInfo can be overridden for a platform by placing a new file in the folder overrides/platorm/whatever. If cmake detects an override, it defines PLATFORM_OVERRIDES that allows the override to create the object.

@wsnipex @fetzerch should be possible with cmake, autotools, right?

### CODE REVIEW

**Comment by popcornmix** [comment]
Thanks. I'll play with this at the weekend.

**Comment by FernetMenta** [comment]
@popcornmix DTS HD decision is supposed to be moved into this class too

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp [hunk 1]
```diff
@@ -418,10 +418,12 @@ void CDVDVideoCodecFFmpeg::SetFilters()
 
   if (mDeintMode != VS_DEINTERLACEMODE_OFF)
   {
-    if(mInt == VS_INTERLACEMETHOD_DEINTERLACE_HALF)
+    if (mInt == VS_INTERLACEMETHOD_DEINTERLACE)
+      filters = FILTER_DEINTERLACE_ANY;
+    if (mInt == VS_INTERLACEMETHOD_DEINTERLACE_HALF)
```
Comment (da-anda) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp]: shouldn't that be an "else if"?

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp]: yes, thanks

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
updated and rebased
@popcornmix could you test this please

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
the usual suspects failed again.

**Comment by Memphiz** [comment]
dropping the vda codec was by intention i suppose? feels a bit burried in this pr ;)

**Comment by FernetMenta** [comment]
@Memphiz it was already dead code and I didn't see any advantage in patching dead code so I dropped it. The plan is still having vtb on OSX.

**Pull request merged**